src/host/hcd.h

```diff
@@ -204,13 +204,11 @@ void hcd_event_xfer_complete(uint8_t dev_addr, uint8_t ep_addr, uint32_t xferred
     .rhport   = 0, // TODO correct rhport
     .event_id = HCD_EVENT_XFER_COMPLETE,
     .dev_addr = dev_addr,
-    .xfer_complete =
-    {
-      .ep_addr = ep_addr,
-      .result  = result,
-      .len     = xferred_bytes
-    }
   };
+  event.xfer_complete.ep_addr = ep_addr;
+  event.xfer_complete.result = result;
+  event.xfer_complete.len = xferred_bytes;
+
```

Comment:
did this not initialize correctly before?

Reply:
I was getting `hcd.h:213:3: sorry, unimplemented: non-trivial designated initializers not supported` and failing on error.

Reply:
ah, oh well lgtm

Reply:
just want to chime in, the warning could be suppressed if we initialize all member by their order.

Reply:
Thats true. I believe for the most part it is in order, just missing the connection struct in hcd_event_t.
So we could add this the mix
```
.connection =
{
    .hub_addr=0,
    .hub_port=0,
    .speed=0
}
```
I can push this change or whatever you think is best

Reply:
no, I think your changes is better, we should only init the one of unioned only.